static/js/components/Group.jsx

```diff
@@ -47,61 +200,378 @@ const Group = () => {
+                </a>
+                &nbsp; for an extensive guide on Alert filters in Fritz.
+              </DialogContentText>
+              <TextField
+                autoFocus
+                required
+                margin="dense"
+                name="filter_name"
+                label="Filter Name"
+                type="text"
+                fullWidth
+                inputRef={register({ required: true, minLength: 3 })}
+              />
+              <FormControl required className={classes.selectEmpty}>
+                <InputLabel name="alert-stream-select-required-label">
+                  Alert stream
+                </InputLabel>
+                <Controller
+                  labelId="alert-stream-select-required-label"
+                  name="filter_stream_id"
+                  as={Select}
+                  defaultValue={0}
+                  control={control}
+                  rules={{ validate: isGoed }}
+                >
+                  {group.streams.map((stream) => (
+                    <MenuItem key={stream.id} value={stream.id}>
+                      {stream.name}
+                    </MenuItem>
+                  ))}
+                </Controller>
+                <FormHelperText>Required</FormHelperText>
+              </FormControl>
+            </DialogContent>
+            <DialogActions>
+              <Button
+                variant="contained"
+                color="primary"
+                className={classes.button_add}
+                type="submit"
+              >
+                Add
+              </Button>
+              <Button autoFocus onClick={handleDialogClose} color="primary">
+                Dismiss
+              </Button>
+            </DialogActions>
+          </form>
+        </Dialog>
+
+        <Dialog
+          fullWidth
+          open={confirmDeleteOpen}
+          onClose={handleConfirmDeleteDialogClose}
+        >
+          <DialogTitle>Delete Group?</DialogTitle>
+          <DialogContent dividers>
+            <DialogContentText>
+              Are you sure you want to delete this Group?
+              <br />
+              Warning! This will delete the group and all of its filters. All
+              source data will be transferred to the Site-wide group.
+            </DialogContentText>
+          </DialogContent>
+
+          <DialogActions>
+            <Button autoFocus onClick={() => setConfirmDeleteOpen(false)}>
+              Dismiss
+            </Button>
+            <Button
+              color="primary"
+              onClick={() => {
+                dispatch(groupsActions.deleteGroup(group.id));
+                setConfirmDeleteOpen(false);
+                history.push("/groups");
+              }}
+            >
+              Confirm
+            </Button>
+          </DialogActions>
+        </Dialog>
       </div>
     );
   }
-  return <div>Loading group</div>;
+  return (
```

Comment:
Slight preference for placing the `return <LoadingPlaceholderThingy />` statement above the main return statement (and inverting the check accordingly if relevant) but this is fine. I like this spinner -- we should adopt this in more places throughout the app!

Reply:
Done